apps/emqx_gateway/src/emqx_gateway_api_clients.erl

```diff
@@ -218,8 +222,8 @@ subscriptions(post, #{
                 of
                     {error, not_found} ->
                         return_http_error(404, "client process not found");
-                    {error, Reason} ->
-                        return_http_error(400, Reason);
+                    {error, {bad_rpc, _} = Reason} ->
```

Comment:
Q: should we keep a catch-all `{error, Reason}` clause?

Reply:
Tbh, I’d rather not handle a `500` situation at all. It’s kind of a contradiction. And also I think any `error` except `not_found` is unexpected.

Reply:
Turns out ... it's needed :D